### PR TITLE
Feature/10508 storage account module with nfsv3

### PR DESCRIPTION
**Pull request by ArtiomMatiom**
Relevant tasks: #10508, #10513 

Extend Storage account module to support NSFv3 and creating containers that support ACL permission

### CODE REVIEW

File: modules/azure/storage_container_nfsv3/main.tf [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    content {
+      scope       = item.value.scope
+      type        = item.value.type
+      permissions = item.value.permissions
+    }
+  }
+}
+
+resource "azurerm_storage_data_lake_gen2_path" "directory" {
+  count = var.directories == null ? 0 : length(var.directories)
```
Comment (Justin-Booij) [inline modules/azure/storage_container_nfsv3/main.tf:31]: Directories is listed as a possible null here but is a required variable, is this correct?

**Review by Justin-Booij — CHANGES_REQUESTED** [review]
Looks good, would like to hear @JeffreyVWT opinion on the IP check since he discussed this before (unless you two already discussed it). Seems fine from my short research though. Just a small change otherwise

File: modules/azure/storage_container_nfsv3/main.tf (hunk 1 above)
Comment (ArtiomMatiom) [reply modules/azure/storage_container_nfsv3/main.tf:31]: Made directories optional

**Comment by ArtiomMatiom** [comment]
Regarding IP, we will be using amazon for now. Talked with Hans. Yet he wanted to look into the VM agents in the long run will create a story for that

**Review by Justin-Booij — APPROVED** [review]

**Pull request merged**